app/forms/validators.py

```diff
@@ -1,33 +1,45 @@
+from __future__ import annotations
+
 import re
 from datetime import datetime, timezone
 from decimal import Decimal, InvalidOperation
+from typing import TYPE_CHECKING, Iterable, List, Mapping, Optional, Sequence, Union
 
 import flask_babel
 from babel import numbers
 from dateutil.relativedelta import relativedelta
 from flask_babel import ngettext
+from flask_wtf import FlaskForm
 from structlog import get_logger
-from wtforms import validators
+from wtforms import Field, IntegerField, StringField, validators
 from wtforms.compat import string_types
 
 from app.forms import error_messages
+from app.forms.fields import DateField, DecimalFieldWithSeparator
 from app.jinja_filters import format_number, get_formatted_currency
 from app.questionnaire.rules import convert_to_datetime
 from app.utilities import safe_content
 
+if TYPE_CHECKING:
+    from app.forms.questionnaire_form import QuestionnaireForm  # pragma: no cover
+
 logger = get_logger()
 
 tld_part_regex = re.compile(
     r"^([a-z]{2,63}|xn--([a-z0-9]+-)*[a-z0-9]+)$", re.IGNORECASE
 )
 email_regex = re.compile(r"^.+@([^.@][^@\s]+)$")
 
+OptionalMessage = Optional[Mapping[str, str]]
+NumType = Union[int, Decimal]
+PeriodType = Mapping[str, int]
+
 
 class NumberCheck:
-    def __init__(self, message=None):
+    def __init__(self, message: str = None):
         self.message = message or error_messages["INVALID_NUMBER"]
 
-    def __call__(self, form, field):
+    def __call__(self, form: FlaskForm, field: IntegerField) -> None:
```

Comment:
It could be a `DecimalField` also, but worth using our classes if it causes no issues.
```suggestion
    def __call__(self, form: FlaskForm, field: Union[DecimalFieldWithSeparator, IntegerFieldWithSeparator]) -> None:
```